### ISSUE TITLE
Compiling a program with no strings gives linking errors

### ISSUE THREAD

**Issue by robryanx** [SerenityOS/jakt#1216]
Any reference at all to a String seems to avoid it, for example `let a: String? = None` is enough to avoid it.
The error given is:
```
/usr/bin/ld: /home/rob/jakt/build/lib/libjakt_main.a(Main.cpp.o): in function `main':
Main.cpp:(.text+0x95): undefined reference to `Jakt::String::copy(Jakt::StringView)'
/usr/bin/ld: /home/rob/jakt/build/lib/libjakt_main.a(Main.cpp.o): in function `Jakt::RefCounted<Jakt::StringStorage>::unref() const':
Main.cpp:(.text._ZNK4Jakt10RefCountedINS_13StringStorageEE5unrefEv[_ZNK4Jakt10RefCountedINS_13StringStorageEE5unrefEv]+0x44): undefined reference to `Jakt::StringStorage::~StringStorage()'
/usr/bin/ld: Main.cpp:(.text._ZNK4Jakt10RefCountedINS_13StringStorageEE5unrefEv[_ZNK4Jakt10RefCountedINS_13StringStorageEE5unrefEv]+0x4d): undefined reference to `Jakt::StringStorage::operator delete(void*)'
/usr/bin/ld: /home/rob/jakt/build/lib/libjakt_main.a(Main.cpp.o): in function `void Jakt::outln<Jakt::Error>(_IO_FILE*, Jakt::StringView&&, Jakt::Error const&)':
Main.cpp:(.text._ZN4Jakt5outlnIJNS_5ErrorEEEEvP8_IO_FILEONS_10StringViewEDpRKT_[_ZN4Jakt5outlnIJNS_5ErrorEEEEvP8_IO_FILEONS_10StringViewEDpRKT_]+0x4b): undefined reference to `Jakt::vout(_IO_FILE*, Jakt::StringView, Jakt::TypeErasedFormatParams&, bool)'
...
```

**Comment by alimpfard** [comment]
This should be fixed as of ec9dc90ea4a8f8ccf503cbc7cc48372e5341b221.

**Comment by robryanx** [comment]
Yeah, looks good now, thanks.
